src/forms/SelectionList/SelectionList.jsx

```diff
@@ -0,0 +1,109 @@
+import { useState, useEffect, useMemo } from 'preact/hooks';
+import PropTypes from 'prop-types';
+
+import Heading from '../../typography/Heading/Heading';
+import CSS from './styles.css';
+
+const SelectionList = ({ className, dataTestId, value, title, options, id, onChange }) => {
+	const [currentValue, setCurrentValue] = useState(null);
+	const delayCallback = 250;
+
+	useEffect(() => {
+		if (value) {
+			setCurrentValue(value);
+			return;
+		}
+		const defaultCheckedOptionIndex = options.findIndex(({ selected }) => selected);
+		setCurrentValue(defaultCheckedOptionIndex !== -1 ? options[defaultCheckedOptionIndex].value : null);
+	}, [value, options]);
+
+	const handleOnClick = ({ selected, label, value: optionValue, children, onClick }, index) => {
+		if (onClick) {
+			setTimeout(() => onClick(optionValue, index), delayCallback);
+		}
+		if (currentValue !== optionValue && onChange) {
+			setTimeout(() => onChange(optionValue, index), delayCallback);
+		}
+		setCurrentValue(optionValue);
+	};
+
+	const result = useMemo(() => {
+		return (
+			<section className={`${CSS.wrapper} ${className}`}>
+				{title && (
+					<Heading className={CSS.title} type="callout">
+						{title}
+					</Heading>
+				)}
+
+				<ul data-testid={`${dataTestId}-select`}>
+					{options.map(({ selected, label, value: optionValue, children, onClick}, index) => {
+						const elementId = `${id}-${index}`;
+						const checked = currentValue === optionValue ? true : false;
+						return (
+							<li 
+								data-testid={`${dataTestId}-option`} 
+								key={index} 
+								data-checked={checked} 
+								className={`${CSS.item} ${checked ? CSS.checked : ''}`} 
+								onClick={event => handleOnClick({ selected, label, value: optionValue, children, onClick }, index)}
+							>
+								<input 
+									type="radio" c
```

Comment:
Please move the `c` at the end of this line to the beginning of the next line

Reply:
🤦 thanks